AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Models/Options/AWSAuthWebUISignInOptions.swift

```diff
@@ -24,8 +25,28 @@ public struct AWSAuthWebUISignInOptions {
     /// are using Auth0, specify the `federationProviderName` as <your_domain>.auth0.com.
     public let federationProviderName: String?
 
-    public init(idpIdentifier: String? = nil, federationProviderName: String? = nil) {
+    /// Start the webUI signin in a private browser session.
+    ///
+    /// Internally this sets the ASWebAuthentication session's prefersEphemeralWebBrowserSession = true.
+    /// Note that this value internally set `prefersEphemeralWebBrowserSession` in ASWebAuthenticationSession.
+    /// As per Apple documentation, Whether the request is honored depends on the user’s default web browser.
+    /// Safari always honors the request.
+    public let preferPrivateSession: Bool
+
+    public init(idpIdentifier: String? = nil,
+                federationProviderName: String? = nil,
+                preferPrivateSession: Bool = false) {
         self.idpIdentifier = idpIdentifier
         self.federationProviderName = federationProviderName
+        self.preferPrivateSession = preferPrivateSession
+    }
+}
+
+extension AuthWebUISignInRequest {
+
+    public static func optionWithPrivateSession() -> AuthWebUISignInOperation.Request.Options {
+        let pluginOptions = AWSAuthWebUISignInOptions(preferPrivateSession: true)
+        let options = AuthWebUISignInOperation.Request.Options(pluginOptions: pluginOptions)
+        return options
     }
```

Comment:
This seems more appropriate as an extension on the options object
```suggestion
extension AWSAuthWebUISignInOptions {
    public static func preferPrivateSession() -> AuthWebUISignInOperation.Request.Options {
        let pluginOptions = AWSAuthWebUISignInOptions(preferPrivateSession: true)
        let options = AuthWebUISignInOperation.Request.Options(pluginOptions: pluginOptions)
        return options
    }
```

What does the call site look like using your code?

Reply:
The intent would be to provide a DX like:

```swift
Amplify.Auth.signInWithWebUI(presentationAnchor: window, options: .preferPrivateSession())
```

So that means you want the extension to be on `AWSAuthWebUISignInOptions`

Reply:
Yes, I updated the code.